pkg/descheduler/controllers/migration/arbitrator/filter.go

```diff
@@ -184,63 +193,64 @@ func (r *Reconciler) forEachAvailableMigrationJobs(listOpts *client.ListOptions,
 	}
 }
 
-func (r *Reconciler) filterExistingPodMigrationJob(pod *corev1.Pod) bool {
-	return !r.existingPodMigrationJob(pod)
+func (f *filter) filterExistingPodMigrationJob(pod *corev1.Pod) bool {
+	return !f.existingPodMigrationJob(pod)
 }
 
-func (r *Reconciler) existingPodMigrationJob(pod *corev1.Pod, expectPhases ...sev1alpha1.PodMigrationJobPhase) bool {
+func (f *filter) existingPodMigrationJob(pod *corev1.Pod, expectedPhaseAndAnnotations ...PhaseAndAnnotation) bool {
 	opts := &client.ListOptions{FieldSelector: fields.OneTermEqualSelector(fieldindex.IndexJobByPodUID, string(pod.UID))}
 	existing := false
-	r.forEachAvailableMigrationJobs(opts, func(job *sev1alpha1.PodMigrationJob) bool {
+	f.forEachAvailableMigrationJobs(opts, func(job *sev1alpha1.PodMigrationJob) bool {
 		if podRef := job.Spec.PodRef; podRef != nil && podRef.UID == pod.UID {
 			existing = true
 		}
 		return !existing
-	}, expectPhases...)
+	}, expectedPhaseAndAnnotations...)
 
 	if !existing {
 		opts = &client.ListOptions{FieldSelector: fields.OneTermEqualSelector(fieldindex.IndexJobPodNamespacedName, fmt.Sprintf("%s/%s", pod.Namespace, pod.Name))}
-		r.forEachAvailableMigrationJobs(opts, func(job *sev1alpha1.PodMigrationJob) bool {
+		f.forEachAvailableMigrationJobs(opts, func(job *sev1alpha1.PodMigrationJob) bool {
 			if podRef := job.Spec.PodRef; podRef != nil && podRef.Namespace == pod.Namespace && podRef.Name == pod.Name {
 				existing = true
 			}
 			return !existing
-		}, expectPhases...)
+		}, expectedPhaseAndAnnotations...)
 	}
 	return existing
 }
 
-func (r *Reconciler) filterMaxMigratingPerNode(pod *corev1.Pod) bool {
-	if pod.Spec.NodeName == "" || r.args.MaxMigratingPerNode == nil || *r.args.MaxMigratingPerNode <= 0 {
+func (f *filter) filterMaxMigratingPerNode(pod *corev1.Pod) bool {
+	if pod.Spec.NodeName == "" || f.args.MaxMigratingPerNode == nil || *f.args.MaxMigratingPerNode <= 0 {
 		return true
 	}
 
 	podList := &corev1.PodList{}
 	listOpts := &client.ListOptions{FieldSelector: fields.OneTermEqualSelector(fieldindex.IndexPodByNodeName, pod.Spec.NodeName)}
-	err := r.Client.List(context.TODO(), podList, listOpts, utilclient.DisableDeepCopy)
+	err := f.client.List(context.TODO(), podList, listOpts, utilclient.DisableDeepCopy)
 	if err != nil {
 		return true
 	}
 	if len(podList.Items) == 0 {
 		return true
 	}
 
-	var expectPhases []sev1alpha1.PodMigrationJobPhase
-	if isPodPrepareMigrating(pod) {
-		expectPhases = append(expectPhases, sev1alpha1.PodMigrationJobRunning)
+	var expectedPhaseAndAnnotations []PhaseAndAnnotation
```

Comment:
As we talked, one scenario is that the built-in plugins call the Filter method of the migration controller to determine whether a new pmj can be created. Running and Pending pmjs should be considered here. At this time, there is no need to consider whether the Pending pmj has been arbitration

Reply:
I will use an annotation to distinguish different scenarios.